AK/Error.h

```diff
@@ -183,14 +183,15 @@ class [[nodiscard]] ErrorOr {
     }
 #endif
 
-    T& value()
-    {
-        return m_value_or_error.template get<T>();
-    }
-    T const& value() const { return m_value_or_error.template get<T>(); }
-
-    ErrorType& error() { return m_value_or_error.template get<ErrorType>(); }
-    ErrorType const& error() const { return m_value_or_error.template get<ErrorType>(); }
+    T& value() & { return m_value_or_error.template get<T>(); }
+    T const& value() const& { return m_value_or_error.template get<T>(); }
+    T&& value() && { return move(m_value_or_error.template get<T>()); }
+    T const&& value() const&& { return move(m_value_or_error.template get<T>()); }
+
+    ErrorType& error() & { return m_value_or_error.template get<ErrorType>(); }
+    ErrorType const& error() const& { return m_value_or_error.template get<ErrorType>(); }
+    ErrorType&& error() && { return move(m_value_or_error.template get<ErrorType>()); }
+    ErrorType const&& error() const&& { return move(m_value_or_error.template get<ErrorType>()); }
```

Comment:
I'm not sure that returning T const&& makes any sense. This declares a const member function on rvalues of type ErrorOr. Which is thus not allowed to modify `this`, and returns a const rvalue? I think to get the expected behavior, you have to move the Error type out of m_value_or_error. So these const rvalue qualified overloads should be deleted.

Reply:
`const&&` is nonsense regardless, this shouldn't exist :)